src/object_storage/objects.rs

```diff
@@ -66,10 +84,13 @@ impl Object {
         Id: AsRef<str>,
         R: Read + Send + 'static,
     {
-        let c_ref = container.into();
-        let c_name = c_ref.to_string();
-        let inner = api::create_object(&session, c_ref, name, body)?;
-        Ok(Object::new(session, inner, c_name))
+        let new_object: NewObject<R> = NewObject::new(
```

Comment:
nit: specific type is probably not needed here, Rust should be smart enough to figure it out.